File: Easy/0892-Surface Area of 3D Shapes/Solution.rs
impl Solution {
    pub fn surface_area(grid: Vec<Vec<i32>>) -> i32 {
        let mut area = 0;

        for i in 0..grid.len() {
            for j in 0..grid[0].len() {
                if grid[i][j] > 0 {
                    area += grid[i][j] * 2 + 1;
                    if i > 0 {
                        area -= grid[i - 1][j].min(grid[i][j]);
                    }
                    if j > 0 {
                        area -= grid[i][j - 1].min(grid[i][j]);
                    }
                }
            }
        }

        area * 2
    }
}

File: Easy/0892-Surface Area of 3D Shapes/Solution.rb
# @param {Integer[][]} grid
# @return {Integer}
def surface_area(grid)
    area = 0

    for i in 0...grid.length
        for j in 0...grid[0].length
            if grid[i][j] > 0
                area += grid[i][j] * 2 + 1
                area -= [grid[i - 1][j], grid[i][j]].min if i > 0
                area -= [grid[i][j - 1], grid[i][j]].min if j > 0
            end
        end
    end

    return area * 2
end
